src/SourceAdapters/StreamAdapter.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Plank\Mediable\SourceAdapters;
+
+use Plank\Mediable\Stream;
+use Plank\Mediable\Exceptions\MediaUpload\ConfigurationException;
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Stream Adapter.
+ *
+ * Adapts a stream object or resource.
+ */
+abstract class StreamAdapter implements SourceAdapterInterface
+{
+    /**
+     * The source object.
+     * @var \Psr\Http\Message\StreamInterface
+     */
+    protected $source;
+
+    /**
+     * The resource.
+     * @var resource|null
+     */
+    protected $resource;
+
+    /**
+     * Constructor.
+     * @param \Psr\Http\Message\StreamInterface|resource $source
+     */
+    public function __construct($source)
+    {
+        if (is_resource($source) && get_resource_type($source) === 'stream') {
+            $this->resource = $source;
+            $this->source = new Stream($source);
```

Comment:
In order to accept both resources and `StreamInterface` object, the resource is being wrapped in a `Plank\Mediable\Stream` object.